### PR TITLE
Exclude Option for protobuf fields

### PR DESCRIPTION
**Pull request by lucasvo**

### CODE REVIEW

File: examples/documents/example.proto [hunk 1]
```diff
@@ -13,6 +17,12 @@ message ExampleDocument {
   int64 value2 = 4;
   bytes value_bytes1 = 5;
   bytes ValueCamelCased = 6;
+  bytes value_not_ignored = 7 [
```
Comment (mikiquantum) [inline examples/documents/example.proto]: Isn't it enough to say that all values are included unless explicitly specified?

File: proofs/proto/proof.proto [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+option go_package = "proofspb";
+option java_multiple_files = true;
+option java_outer_classname = "ProofProto";
+option java_package = "com.proofs";
+
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FieldOptions {
+  bool exclude_from_tree = 2862100;
```
Comment (mikiquantum) [inline proofs/proto/proof.proto]: Just curious, what is this long number for?

File: proofs/tree.go [hunk 3]
```diff
@@ -333,53 +310,138 @@ func getPropertyNameFromProtobufTag(tag string) (name string, err error) {
...
+		fieldDescriptor := messageDescriptor.Field[i]
+		fV := reflect.ValueOf(fieldDescriptor).Elem()
+		fType := fV.Type()
+		fieldName := *fieldDescriptor.Name
+		for i := 0; i < fV.NumField(); i++ {
+			field := fV.Field(i)
+			if fType.Field(i).Name != "Options" {
+				continue
+			}
+			if proto.HasExtension(field.Interface().(proto.Message), proofspb.E_ExcludeFromTree) {
```
Comment (mikiquantum) [inline proofs/tree.go]: If we already find that the field has the `ExcludeFromTreeExtension`, isn't it safe to assume that it should be excluded? (following on a previous comment)
So might not need to get the Extension value.

File: proofs/proto/proof.proto (hunk 2 above)
Comment (lucasvo) [reply proofs/proto/proof.proto]: Arbitrary, not to clash with any existing options.

File: examples/documents/example.proto (hunk 1 above)
Comment (lucasvo) [reply examples/documents/example.proto]: Yes, but because it's technically legal syntax, I wanted to add a test case

**Pull request merged**